flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -347,43 +351,88 @@ public static String getTaskManagerShellCommand(
 			boolean hasKrb5,
 			Class<?> mainClass) {
```

Comment:
Did you consider using this method also for the JobManager / ApplicationMaster container invocation in `AbstractYarnClusterDescriptor.java` ? I'm not sure if my approach is the right one, but there is a lot of shared code between the two.